src/auto_scheduler/search_task.cc

```diff
@@ -90,6 +90,22 @@ HardwareParams HardwareParamsNode::GetDefaultHardwareParams(const Target& target
     int max_vthread_extent = warp_size / 4;
     return HardwareParams(-1, 16, 64, max_shared_memory_per_block, max_local_memory_per_block,
                           max_threads_per_block, max_vthread_extent, warp_size);
+  } else if (target->kind->device_type == kDLOpenCL) {
+    if (target->GetAttr<String>("device", "") == "mali") {
+      // We can not use device api to get attr like CUDA
+      // because like Mali target is normally on the remote machine
```

Comment:
nit: maybe we can tell users how to get the Mali hardware parameters manually in the tutorial.

```suggestion
      // We cannot use device API to get hardware attributes like CUDA,
      // because Mali target is normally on the remote machine.
```